src/test/kotlin/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtSharedTableBackupManagerImplTest.kt

```diff
@@ -0,0 +1,160 @@
+ */
+package com.salesforce.dynamodbv2.mt.mappers.sharedtable.impl
+
+import com.amazonaws.services.dynamodbv2.model.AttributeDefinition
+import com.amazonaws.services.dynamodbv2.model.AttributeValue
+import com.amazonaws.services.dynamodbv2.model.CreateBackupRequest
+import com.amazonaws.services.dynamodbv2.model.GetItemRequest
+import com.amazonaws.services.dynamodbv2.model.KeySchemaElement
+import com.amazonaws.services.dynamodbv2.model.KeyType
+import com.amazonaws.services.dynamodbv2.model.PutItemRequest
+import com.amazonaws.services.dynamodbv2.model.ScalarAttributeType
+import com.amazonaws.services.s3.AmazonS3
+import com.amazonaws.services.s3.AmazonS3ClientBuilder
+import com.amazonaws.services.s3.model.CreateBucketRequest
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import com.google.common.collect.Lists
+import com.salesforce.dynamodbv2.dynamodblocal.AmazonDynamoDbLocal
+import com.salesforce.dynamodbv2.mt.context.MtAmazonDynamoDbContextProvider
+import com.salesforce.dynamodbv2.mt.context.impl.MtAmazonDynamoDbContextProviderThreadLocalImpl
+import com.salesforce.dynamodbv2.mt.mappers.CreateTableRequestBuilder
+import com.salesforce.dynamodbv2.mt.mappers.MtAmazonDynamoDb.TenantTable
+import com.salesforce.dynamodbv2.mt.mappers.sharedtable.SharedTableBuilder
+import com.salesforce.dynamodbv2.mt.mappers.sharedtable.CreateMtBackupRequest
+import com.salesforce.dynamodbv2.mt.mappers.sharedtable.MtBackupManager
+import com.salesforce.dynamodbv2.mt.mappers.sharedtable.MtBackupMetadata
+import com.salesforce.dynamodbv2.mt.mappers.sharedtable.RestoreMtBackupRequest
+import com.salesforce.dynamodbv2.mt.mappers.sharedtable.Status
+import com.salesforce.dynamodbv2.mt.repo.MtTableDescriptionRepo
+import com.salesforce.dynamodbv2.testsupport.ItemBuilder.HASH_KEY_FIELD
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+internal class MtSharedTableBackupManagerImplTest {
+
+    companion object {
+        val REGION = "us-east-1"
+        val MT_CONTEXT: MtAmazonDynamoDbContextProvider = MtAmazonDynamoDbContextProviderThreadLocalImpl()
+
+        val bucket: String = "test-basic-backup-create"
+        val dynamo = AmazonDynamoDbLocal.getAmazonDynamoDbLocal()
+
+        val sharedTableBinaryHashKey = SharedTableBuilder.builder()
+                .withAmazonDynamoDb(dynamo)
+                .withContext(MT_CONTEXT)
+                .withBackupSupport(REGION, bucket)
+                .withTruncateOnDeleteTable(true)
+                .withBinaryHashKey(true)
+                .build()
+
+        val backupManager = sharedTableBinaryHashKey.backupManager
+
+        var s3: AmazonS3? = null
+
+        @BeforeAll
+        @JvmStatic
+        internal fun beforeAll() {
+            // TODO: Make this test operate against mock s3 in an autogenerated bucket
```

Comment:
grrr, I did spend a good amount of time trying to use a mock s3 implementation that zero-importer uses in its unit tests, but ran into maven dependency hell. The local dynamo db jar pulls in dependencies for a super old jetty server jar, which collides with the mock s3 newer jetty server jar, and blows up. Tried a few things unsuccessfully, and got frustrated. 

Debating switching the unit tests over to operate against dynamo in docker, so the JVM doesn't have to pull in ancient jetty jars, but that's non-ideal given the number of test cases we have using local dynamo and how many local flows would get disrupted. 

I guess I can go the other way, and setup a mock s3 in a local docker like Brian has in the local universe, and have our test suite do a docker compose and spin up a local s3? Thoughts?